spec/parser/options_spec.rb

```diff
@@ -116,7 +116,9 @@ def remaining
       expected = "Unknown switches \"--baz\""
       expected << "\nDid you mean?  \"--bar\"" if Thor::Correctable
 
-      expect { check_unknown! }.to raise_error(Thor::UnknownArgumentError, expected)
+      expect { check_unknown! }.to raise_error(Thor::UnknownArgumentError) do |error|
+        expect(error.to_s).to eq(expected)
+      end
```

Comment:
@voxik pointed out [elsewhere](https://github.com/rails/thor/pull/782#discussion_r823669341):

> I wonder, shouldn't rather be the `Thor::Correctable` line above dropped instead? But that would probably not be backward compatible with older rspec-mock 🤔

I can validate that removing that line instead does fix it for ruby 3.0.

> This is caused by [rspec/rspec-expectations#1339](https://github.com/rspec/rspec-expectations/pull/1339) just for the context.

But I don't think this is the same issue addressed by that PR, because https://github.com/rspec/rspec-expectations/pull/1339 was [included in  rspec-expectations v3.11.0](https://github.com/rspec/rspec-expectations/commit/44f9a2448c53d71eaea1f45ecb3473048bacb0de), and using that version, you still get an error in main.

I'm neutral on which way to fix it (just want the CI to be green 😀 )

Reply:
Reported as https://github.com/rspec/rspec-expectations/issues/1362

Reply:
@voxik can you clarify for me - is there any action needed on this PR or is it good to go in your opinion?